src/pj_gridinfo.c

```diff
@@ -568,10 +579,23 @@ static int pj_gridinfo_init_ntv2( projCtx ctx, PAFile fid, PJ_GRIDINFO *gilist )
         else
         {
             gi = (PJ_GRIDINFO *) pj_malloc(sizeof(PJ_GRIDINFO));
+            if (!gi) {
+                pj_dalloc(ct);
+                pj_gridinfo_free(ctx, gilist);
+                pj_ctx_set_errno(ctx, ENOMEM);
+                return 0;
+            }
             memset( gi, 0, sizeof(PJ_GRIDINFO) );
```

Comment:
You can eliminate this by switching to `pj_calloc` in line 581. While slightly outside the scope of this PR, it is still about memory handling, and in general, I think using calloc when needing a 0-initialized buffer is the right thing to do.